src/window-rules.c

```diff
@@ -67,6 +67,8 @@ view_matches_criteria(struct window_rule *rule, struct view *view)
 			return false;
 		}
 		return match_glob(rule->title, title);
+	} else if (rule->window_type >= 0 && view->impl->contains_window_type) {
+		return view->impl->contains_window_type(view, rule->window_type);
```

Comment:
Is it intentional that `window_type` conditions are ignored if any other conditions are provided?

Reply:
Likely not.

There should be some `bool matches = false` on top and all conditions should result in `matches |= condition` and at the end we return `matches`. However, a condition being absent from the rule should not changes the `matches` bool.

Reply:
see 327dd48

Reply:
Thanks, should be fixed in https://github.com/labwc/labwc/pull/1490/commits/33cb3d47f8cab58166ce216dc4470afa85e7ec78

Reply:
Thanks, should be fixed in https://github.com/labwc/labwc/pull/1490/commits/33cb3d47f8cab58166ce216dc4470afa85e7ec78

Reply:
First of all, sorry for the delay in this PR.

It seems like #1731 will get merged first (as it doesn't add the internal rules), @grisha128 are you fine with my Co-Authored-By suggestion from https://github.com/labwc/labwc/pull/1731#issuecomment-2066839523 ?

When #1731 is merged, would you be fine with rebasing your work on top? That should also solve the "no way for users to override the internal rules" issue.

Reply:
> First of all, sorry for the delay in this PR.

No problem, all is fine ;)

> It seems like https://github.com/labwc/labwc/pull/1731 will get merged first (as it doesn't add the internal rules), @grisha128 are you fine with my Co-Authored-By suggestion from https://github.com/labwc/labwc/pull/1731#issuecomment-2066839523 ?

I guess that's okay!

> When https://github.com/labwc/labwc/pull/1731 is merged, would you be fine with rebasing your work on top? That should also solve the "no way for users to override the internal rules" issue.

Will do, thank you <3